packages/http-client/src/http-client.spec.ts

```diff
@@ -207,6 +208,46 @@ describe('generic http client', () => {
       jest.restoreAllMocks();
     });
 
+    function middleWareBuilder(appendedText: string): Middleware<HttpResponse> {
+      return (options: MiddlewareInOut<HttpResponse>) => {
+        const wrapped = () =>
+          options.fn().then(res => {
+            res.data = res.data + appendedText;
+            return res;
+          });
+
+        return {
+          ...options,
+          fn: wrapped
+        };
+      };
+    }
+
+    it('attaches one middleware', async () => {
+      nock('https://example.com').get(/.*/).reply(200, 'Initial value.');
+      const myMiddleware = middleWareBuilder('Middleware One.');
+
+      const response = await executeHttpRequest(httpsDestination, {
+        middleware: [myMiddleware],
+        method: 'get'
+      });
+      expect(response.data).toEqual('Initial value.Middleware One.');
+    });
+
+    it('attaches multiple middleware in the expected order', async () => {
+      nock('https://example.com').get(/.*/).reply(200, 'Initial value.');
+      const myMiddlewareOne = middleWareBuilder('Middleware One.');
+      const myMiddlewareTwo = middleWareBuilder('Middleware Two.');
+
+      const response = await executeHttpRequest(httpsDestination, {
+        middleware: [myMiddlewareOne, myMiddlewareTwo],
+        method: 'get'
+      });
```

Comment:
I would 
```suggestion
      const myMiddlewareTwo = middleWareBuilder('Middleware Two.');
      const myMiddlewareOne = middleWareBuilder('Middleware One.');

      const response = await executeHttpRequest(httpsDestination, {
        middleware: [myMiddlewareOne, myMiddlewareTwo],
        method: 'get'
      });
```
[pp] I would swap the creation of the middlewares. This would make it clear that the creation is not relevant for the execution order (and break the test if this changes).